### ISSUE TITLE
Maya Tool - Solver UI Error Graph

### ISSUE THREAD

**Issue by david-cattermole** [david-cattermole/mayaMatchMoveSolver#12]
Currently we display the overall solved error in the Script Editor, this is not ideal.

To allow better understanding and solver debugging for users, we should show the error (deviation) overall, and for each marker.

Ideas:
- Give feedback to user about solve quality.
- We can use animCurves to show error over time! We should create attributes on the Collection which keyframe error over time.
- We can create per-marker errors as keyframes on specially named attributes.
- After each successful solve, the previous errors are removed and replaced with the new Python ``SolveResult`` objects. This will store various debug information, so developers can debug scene files, and save the information in the Maya scene.

Ideally, this Error Graph would be visible inside the Solver UI.

**Comment by david-cattermole** [comment]
We have added a live deviation read-out in the Solver UI, for Markers and Cameras. This read-out will constantly compute the 2D-to-3D reprojection distance, in pixels.

Notes to address;
- Image resolution must come from the Camera class, which should read it from the attached image plane. If a resolution cannot be found the API classes should return None. It is up to the user-code to pick a default. We should probably use "2048" as an approximate image width if none is found, or "4096"?
- We should be able to turn on/off the display of the deviation and marker weight columns in the Solver UI. This should be controlled by a "View" menu, and the values should be stored in the Maya scene, not dependent on the Collection.
- We need to write deviation errors to animCurves on the respective Markers.
- By default, the Solver UI should *not* compute the deviation live, but instead use the cached values.
- If a live deviation is used, we must update the Solver UI using a callback when the user changes the current frame, but we must also make sure this "update" is turned off while solving.

**Comment by david-cattermole** [comment]
I've addressed a number of the previous points in the latest commit (7062db1) on ```tools_issue12``` branch.

Upon further thought I've decided not to allow "live" deviation computation. I feel this may be confusing to users, and likey slow down the performance of the UI - using cached deviation computation is already slow enough.

Features:
- Added View menu to turn on/off deviation, marker weight (and other) columns in the Solver UI.
- Marker and Solve Deviation are set as keyframe animCurves on the Marker and Collection nodes; these are viewable in the Solver UI or the Maya Graph Editor.
 
Still to address:
- Image resolution must come from the Camera class, which should read it from the attached image plane. If a resolution cannot be found the API classes should return None. It is up to the user-code to pick a default. We should probably use "2048" as an approximate image width if none is found, or "4096"?
